### DISCUSSION TITLE
properly closing live query

### DISCUSSION THREAD

**Discussion by donl** [appy-one/acebase#108] [Q&A] [answered]
When a live query is no longer needed, what is the correct way to unsubscribe and destroy it?

**Comment by donl** [comment]
I'm assuming calling ```.stop()``` on the returned ```Promise``` ?

Maybe a better followup question -- Is it possible to use the ``exclude``, ``include``, ``child_objects``, and/or``snapshots``  options with a realtime query?

**Reply by donl** [reply]
Following the code referenced below, I see how to pass options to a live query and process the results.

https://github.com/appy-one/acebase/blob/76a20b66e2dac08512240f62705d0a3d3e284ae1/spec/query.spec.js#L112

Stopping the live query required calling the equivalent ``.off()`` for each ``.on()`` that was used.

**Comment by appy-one** [comment]
Good question. I checked the code, it seems that stopping a realtime query isn't implemented yet!
Usage of the `exclude`, `include`, `child_objects` and `snapshots` should work.

**Comment by appy-one** [comment, marked as answer]
I forgot to mention last week that you should now be able to stop the queries using `query.stop()`, see #113!
